application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/Transfer.java

```diff
@@ -31,16 +33,20 @@ public class Transfer {
 
   private final List<Edge> edges;
 
-  public Transfer(int toStop, List<Edge> edges) {
+  private final EnumSet<StreetMode> modes;
+
+  public Transfer(int toStop, List<Edge> edges, EnumSet<StreetMode> modes) {
     this.toStop = toStop;
     this.edges = edges;
     this.distanceMeters = (int) edges.stream().mapToDouble(Edge::getDistanceMeters).sum();
+    this.modes = modes;
```

Comment:
We probably want to make this immutable. EnumSet are not.

Reply:
I changed the `getModes` method to clone the enum set